### ISSUE TITLE
Only the first UIKitView in a Box is displayed properly on iOS

### ISSUE THREAD

**Issue by brewin** [JetBrains/compose-multiplatform#4004]
**Describe the bug**
Multiple UIKitViews cannot be displayed in a Box on iOS. Only the first UIKitView is displayed properly.

**Affected platforms**
- iOS

**Versions**
- Kotlin version*: 1.9.21
- Compose Multiplatform version*: 1.6.0-dev1296
- OS version(s)* (required for Desktop and iOS issues): iOS 17.0
- OS architecture (x86 or arm64): arm64

**To Reproduce**

Common
```Kotlin
@Composable
expect fun DemoView()
```
iOS
```Kotlin
@OptIn(ExperimentalForeignApi::class)
@Composable
actual fun DemoView() {
    Box(modifier = Modifier.fillMaxSize().background(Color.Yellow)) {
        UIKitView(
            factory = {
                UIView().apply {
                    backgroundColor = UIColor.blueColor
                }
            },
            modifier = Modifier.size(100.dp),
        )
        UIKitView(
            factory = {
                UIView().apply {
                    backgroundColor = UIColor.redColor
                }
            },
            modifier = Modifier.size(100.dp).offset(50.dp, 50.dp),
        )
    }
}
```
Android
```Kotlin
@Composable
actual fun DemoView() {
    Box(modifier = Modifier.fillMaxSize().background(Color.Yellow)) {
        AndroidView(
            factory = { context ->
                View(context).apply {
                    setBackgroundColor(Color.Blue.toArgb())
                }
            },
            modifier = Modifier.size(100.dp)
        )
        AndroidView(
            factory = { context ->
                View(context).apply {
                    setBackgroundColor(Color.Red.toArgb())
                }
            },
            modifier = Modifier.size(100.dp).offset(50.dp, 50.dp)
        )
    }
}
```

**Expected behavior**
On iOS, only the blue square is displayed properly, with the other square appearing to be rendered behind it and without color. I expect a red square to be displayed overlapping a blue square, like on Android.

**Comment by m-sasha** [comment]
On 2nd thought maybe we can do this. It does work on the desktop:

```
fun main() = singleWindowApplication {
    Column(Modifier.fillMaxSize()) {
        SwingPanel(
            modifier = Modifier.height(200.dp).fillMaxWidth(),
            factory = {
                JPanel().also {
                    it.background = java.awt.Color.RED
                }
            }
        )
        SwingPanel(
            modifier = Modifier.height(200.dp).fillMaxWidth(),
            factory = {
                JPanel().also {
                    it.background = java.awt.Color.YELLOW
                }
            }
        )
        SwingPanel(
            modifier = Modifier.height(200.dp).fillMaxWidth(),
            factory = {
                JPanel().also {
                    it.background = java.awt.Color.GREEN
                }
            }
        )
    }
}
```

<img width="801" alt="image" src="https://github.com/JetBrains/compose-multiplatform/assets/5230206/38965feb-aeaf-460c-ad96-2ae0a0395165">

**Comment by m-sasha** [comment]
Hmm, looks like there are several issues here. I'm getting this:
```
    Box(modifier = Modifier.fillMaxSize().background(Color.Yellow)) {
        UIKitView(
            factory = {
                UIView().apply {
                    backgroundColor = UIColor.blueColor
                }
            },
            modifier = Modifier.size(100.dp),
        )
        UIKitView(
            factory = {
                UIView().apply {
                    backgroundColor = UIColor.redColor
                }
            },
            modifier = Modifier.size(100.dp).offset(50.dp, 50.dp),
        )
    }
```

<img width="427" alt="image" src="https://github.com/JetBrains/compose-multiplatform/assets/5230206/4d4be179-d082-497d-a7de-7525e5f758a4">

Reordering the `size` and `offset` modifiers I get this:

<img width="427" alt="image" src="https://github.com/JetBrains/compose-multiplatform/assets/5230206/1a7c7dbd-bbc6-4220-819f-6f23d817d10f">

So there are at least two problems:
1. It seems the native views are added in reverse order. @dima-avdeev-jb is taking care of this.
2. The order of modifiers somehow matters, although it seems it shouldn't.

Also @brewin are the screenshots above representative of what you're seeing? Because with the code you provided I do see a white rectangle for the 2nd view, whereas you mentioned that only the 1st view is visible.

**Comment by brewin** [comment]
@m-sasha Yes, originally I didn't see the white square. I edited the issue to add the yellow background which makes the white square visible.

Setting the blue square translucent shows the issue with the order.

```Kotlin
    Box(modifier = Modifier.fillMaxSize().background(Color.Yellow)) {
        UIKitView(
            factory = {
                UIView().apply {
                    backgroundColor = UIColor.blueColor.colorWithAlphaComponent(0.5)
                }
            },
            modifier = Modifier.size(100.dp),
            background = Color.Transparent,
        )
        UIKitView(
            factory = {
                UIView().apply {
                    backgroundColor = UIColor.redColor
                }
            },
            modifier = Modifier
                .offset(50.dp, 50.dp)
                .size(100.dp),
            background = Color.Transparent,
        )
    }
```
<img src="https://github.com/JetBrains/compose-multiplatform/assets/1248685/0c27dbd4-c31f-451e-b617-2111fbabd07a" width="50%">


When `size` comes before `offset`, the first square turns purple. It seems the colors are blended before the offset occurs.

<img src="https://github.com/JetBrains/compose-multiplatform/assets/1248685/3e7a3e87-205d-4393-9e8e-b16972e6d485" width="50%">

**Comment by dima-avdeev-jb** [comment]
Fixed in 1.6.0-rc01 and later

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-4004">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
